### PR TITLE
Support character array in read function

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Towards: https://github.com/lfortran/lfortran/issues/2324

### CODE REVIEW

File: integration_tests/file_09.f90 [hunk 1]
```diff
@@ -0,0 +1,22 @@
+program file_06
```
Comment (certik) [inline integration_tests/file_09.f90:1]:
```suggestion
program file_09
```

File: src/libasr/runtime/lfortran_intrinsics.c [hunk 2]
```diff
@@ -2036,6 +2036,36 @@ LFORTRAN_API void _lfortran_read_array_double(double *p, int array_size, int32_t
...
+
+    bool unit_file_bin;
+    FILE* filep = get_file_pointer_from_unit(unit_num, &unit_file_bin);
+    if (!filep) {
+        printf("No file found with given unit\n");
+        exit(1);
+    }
+
+    for (int i = 0; i < array_size; i++) {
+        int n = 1; // TODO
```
Comment (certik) [inline src/libasr/runtime/lfortran_intrinsics.c:2059]: Put here what exactly is supposed to be fixed in the future.

**Review by certik — APPROVED** [review]
Looks great!

File: src/libasr/runtime/lfortran_intrinsics.c (hunk 2 above)
Comment (Thirumalai-Shaktivel) [reply src/libasr/runtime/lfortran_intrinsics.c]: Done!

File: integration_tests/file_09.f90 (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply integration_tests/file_09.f90]: Done!

**Review by certik — APPROVED** [review]
I think this looks great.

**Pull request merged**